napari/utils/events/evented_model.py

```diff
@@ -309,15 +323,27 @@ def update(
 
         with self.events.blocker() as block:
             for key, value in values.items():
-                field = getattr(self, key)
-                if isinstance(field, EventedModel) and recurse:
-                    field.update(value, recurse=recurse)
-                else:
-                    setattr(self, key, value)
+                setattr(self, key, value)
 
         if block.count:
             self.events(Event(self))
 
+    def _update_inplace(self, other):
+        self.update(other)
+
+    def _validate(self, new_values):
+        """
+        validate values against the current model. This differs from pydantic's public
+        validate method because it won't return an instance of Self (expensive for evented models)
+        """
+        # __dict__ is faster than dict() and preserves inner types (see below)
+        values = self.__dict__.copy()
+        values.update(new_values)
+        values, _, error = validate_model(self.__class__, values)
+        if error:
+            raise error
```

Comment:
proper reraising error during validation should be tested, as we use pydantic to get proper errors about wrong values.

Reply:
Can you elaborate what you mean by proper reraising?

Reply:
Code coverage shows that this line is not covered in tests. And, from my perspective, it is required to be sure on the level of test checks if a proper exception is raised.

Reply:
added test.